import 'package:flutter/foundation.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:riverpod_annotation/riverpod_annotation.dart';

import 'app_exception.dart';

part 'error_logger.g.dart';

class ErrorLogger {
  void logError(
    Object error,
    StackTrace? stackTrace,
  ) {
    // * This can be replaced with a call to a crash reporting tool of choice
    debugPrint('$error, $stackTrace');
  }

  void logAppException(AppException exception) {
    // * This can be replaced with a call to a crash reporting tool of choice
    debugPrint('$exception');
  }
}

@riverpod
Future<void> triggerError(Ref ref) async {
  throw Exception('Simulated error for logging test');
}

@riverpod
ErrorLogger errorLogger(Ref ref) {
  return ErrorLogger();
}
